target/linux/ar71xx/image/generic.mk

```diff
@@ -790,6 +790,19 @@ define Device/bhr-4grv2
 endef
 TARGET_DEVICES += bhr-4grv2
 
+define Device/yun
+  DEVICE_TITLE := Arduino Yun
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-storage kmod-spi-gpio kmod-serial-8250 kmod-ath9k
```

Comment:
Your patch says "migrate... from legacy to generic", so this should go into a separate patch, with a proper description. Please, don't hide in your patch changes which are out of scope of the patch.